napari_animation/easing.py

```diff
@@ -0,0 +1,285 @@
+"""
+The easing functions here were adapted from
+https://gist.github.com/zeffii/c1e14dd6620ad855d81ec2e89a859719
+"""
+
+from math import sqrt, pow, sin, cos, pi
+from enum import Enum
+
+tau = pi * 2
+
+
+class Easing(Enum):
+    """Easing: easing function to use for a transition.
+
+    Selects a preset easing function
+            * linear: linear interpolation between start and endpoint.
+            * quadratic: quadratic easing in and out.
+                Modeled after the piecewise quadratic
+                y = (1/2)((2x)^2)             ; [0, 0.5)
+                y = -(1/2)((2x-1)*(2x-3) - 1) ; [0.5, 1]
+            * cubic: cubic easing in and out.
+                Modeled after the piecewise cubic
+                y = (1/2)((2x)^3)       ; [0, 0.5)
+                y = (1/2)((2x-2)^3 + 2) ; [0.5, 1]
+            * quintic: quintic easing in and out.
+                Modeled after the piecewise quintic
+                y = (1/2)((2x)^5)       ; [0, 0.5)
+                y = (1/2)((2x-2)^5 + 2) ; [0.5, 1]
+            * sine: sinusoidal easing in and out.
+                Modeled after half sine wave
+                y = 0.5 * (1 - cos(x * pi))
+            * circular: circular easing in and out.
+                Modeled after the piecewise circular function
+                y = (1/2)(1 - sqrt(1 - 4x^2))           ; [0, 0.5)
+                y = (1/2)(sqrt(-(2x - 3)*(2x - 1)) + 1) ; [0.5, 1]
+            * exponential: exponential easing in and out.
+                Modeled after the piecewise exponential
+                y = (1/2)2^(10(2x - 1))         ; [0,0.5)
+                y = -(1/2)*2^(-10(2x - 1))) + 1 ; [0.5,1]
+            * elastic: elastic easing in and out.
+                Modeled after the piecewise exponentially-damped sine wave:
+                y = (1/2)*sin(13pi/2*(2*x))*pow(2, 10 * ((2*x) - 1))      ; [0, 0.5)
+                y = (1/2)*(sin(-13pi/2*((2x-1)+1))*pow(2,-10(2*x-1)) + 2) ; [0.5, 1]
+            * back: back easing in and out.
+                Modeled after the piecewise overshooting cubic function:
+                y = (1/2)*((2x)^3-(2x)*sin(2*x*pi))           ; [0, 0.5)
+                y = (1/2)*(1-((1-x)^3-(1-x)*sin((1-x)*pi))+1) ; [0.5, 1]
+            * bounce: bounce easing in and out.
+    """
+    LINEAR = 1
+    QUADRATIC = 2
+    CUBIC = 3
+    QUINTIC = 4
+    SINE = 5
+    CIRCULAR = 6
+    EXPONENTIAL = 7
+    ELASTIC = 8
+    BACK = 9
+    BOUNCE = 10
```

Comment:
Can you make these of the form `LINEAR='linear'` as it means people can then use string for Enum along with above change, see https://stackoverflow.com/a/63028809

Reply:
fine to keep all caps on the enum part and go lower case on the string part

Reply:
done and done! thanks!